server/src/main/java/org/opensearch/search/aggregations/bucket/nested/NestedAggregator.java

```diff
@@ -130,6 +142,29 @@ public void collect(int parentDoc, long bucket) throws IOException {
         }
     }
 
+    static Tuple<Integer, Integer> getChildAndParentId(BitSet parentDocs, DocIdSetIterator childDocs, int parentDoc) throws IOException {
+        int currentParentDoc;
+        int prevParentDoc = parentDocs.prevSetBit(parentDoc);
+        if (prevParentDoc == -1) {
+            currentParentDoc = parentDocs.nextSetBit(0);
```

Comment:
@kkewwei, correct me if I'm wrong, but this branch would only get reached if:
1. `parentDoc` already points to the first parent doc, in which case `currentParentDoc == parentDoc`.
2. `parentDoc` points to a child of the first parent, in which case `currentParentDoc > parentDoc`.

If `parentDoc` *doesn't* point to a parent document, I believe there's a mistake somewhere else in the code.

Reply:
@msfroh, the two points are right.  (when aggregation, nest2 is outer path, nest1 is inner path)
first case:
<img width="472" alt="image" src="https://github.com/opensearch-project/OpenSearch/assets/16730433/84146e7b-de41-4b80-9664-620094b80558">
parentDoc can be 1,3,5

second case:
<img width="609" alt="image" src="https://github.com/opensearch-project/OpenSearch/assets/16730433/fb40477d-3f44-488c-b266-7dcc4f682cbe">
parentDoc can be 0,3,6

Reply:
In the first case, field `nest2` is the parent doc of `nest1`(in which case currentParentDoc == parentDoc)
In the second case, field `nest2` and field `nest1` is independant,  `nest2` is behind/after`nest1 `.

Reply:
Assuming my other comment is correct, it would probably be a good idea to add some comments here and rename some variables to explain the situation. 

Maybe the `parentDoc` parameter should be called `parentAggDoc`. If it's a top-level aggregation, it's reasonable that it would be the same as the block's parent doc. If it's also a `nested` aggregation, then it may be a sibling (in the block).

So, while the input parameter passed in here is the parent aggregation's current doc (which may or may not be a block-level parent doc), what we output  is a tuple consisting of the current block-level parent doc (the parent of the parameter doc), and the next matching child doc (hopefully under this parent) for the aggregation (according to the child doc iterator). 

If the next matching child doc is *not* under the block-parent of the current agg-parent's doc, then we return `NO_MORE_DOCS` as a signal that we have no matches for the current parent agg bucket. Actually, I think that part is unnecessary because of the condition on the for-loop on line 137. So, I think you can remove the part from line 168-170.

@kkewwei -- have I understood correctly?